auth/tenant_mgt.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import "errors"
+
+// Tenant represents a tenant in a multi-tenant application.
+//
+// Multi-tenancy support requires Google Cloud's Identity Platform (GCIP). To learn more about GCIP,
+// including pricing and features, see https://cloud.google.com/identity-platform.
+//
+// Before multi-tenancy can be used in a Google Cloud Identity Platform project, tenants must be
+// enabled in that project via the Cloud Console UI.
+//
+// A tenant configuration provides information such as the display name, tenant identifier and email
+// authentication configuration. For OIDC/SAML provider configuration management, TenantClient
+// instances should be used instead of a Tenant to retrieve the list of configured IdPs on a tenant.
```

Comment:
Minor thing: is this `IdPs` or `IDPs`? :)

Reply:
Both forms are in popular use. I've seen both in documentation.

e.g. https://support.google.com/a/answer/60224?hl=en uses IdP